local-tests/setup/tinny-operations.ts

```diff
@@ -146,57 +149,69 @@ export const runTestsParallel = async ({
-        try {
-          console.log(
-            `\x1b[90m[runTestsParallel] Attempt ${attempts + 1} for ${
-              testIndex + 1
-            }. ${testName}...\x1b[0m`
-          );
+  const runTest = async (
+    [testName, testFunction]: [string, any],
+    testIndex: number
+  ): Promise<string> => {
+    const maxAttempts = devEnv.processEnvs.MAX_ATTEMPTS;
+    let attempts = 0;
+    let testPassed = false;
+
+    while (attempts < maxAttempts && !testPassed) {
+      const startTime = performance.now();
+      try {
+        console.log(
+          `\x1b[90m[runTestsParallel] Attempt ${attempts + 1} for ${
+            testIndex + 1
+          }. ${testName}...\x1b[0m`
+        );
 
-          // @ts-ignore
-          await testFunction(devEnv);
-          testPassed = true;
+        // @ts-ignore
+        await testFunction(devEnv);
+        testPassed = true;
 
+        const endTime = performance.now();
+        const timeTaken = (endTime - startTime).toFixed(2);
+        console.log(
+          `\x1b[32m✔\x1b[90m ${
+            testIndex + 1
+          }. ${testName} - Passed (${timeTaken} ms)\x1b[0m`
+        );
+        return `${testName} (Passed in ${timeTaken} ms)`;
+      } catch (error) {
+        if (error.message === 'LIT_IGNORE_TEST') {
+          return `${testName} (Skipped)`;
+        }
+        attempts++;
+
+        if (attempts >= maxAttempts) {
           const endTime = performance.now();
           const timeTaken = (endTime - startTime).toFixed(2);
-          console.log(
-            `\x1b[32m✔\x1b[90m ${
+          console.error(
+            `\x1b[31m✖\x1b[90m ${
               testIndex + 1
-            }. ${testName} - Passed (${timeTaken} ms)\x1b[0m`
+            }. ${testName} - Failed after ${maxAttempts} attempts (${timeTaken} ms)\x1b[0m`
           );
-          return `${testName} (Passed in ${timeTaken} ms)`;
-        } catch (error) {
-          if (error.message === 'LIT_IGNORE_TEST') {
-            return `${testName} (Skipped)`;
-          }
-          attempts++;
-
-          // wait for at least 5 seconds
-          if (attempts >= maxAttempts) {
-            const endTime = performance.now();
-            const timeTaken = (endTime - startTime).toFixed(2);
-            console.error(
-              `\x1b[31m✖\x1b[90m ${
-                testIndex + 1
-              }. ${testName} - Failed after ${maxAttempts} attempts (${timeTaken} ms)\x1b[0m`
-            );
-            console.error(`\x1b[31m❌Error:\x1b[90m ${error}\x1b[0m`);
-            return `${testName} (Failed in ${timeTaken} ms) - Error: ${error}`;
-          }
+          console.error(`\x1b[31m❌Error:\x1b[90m ${error}\x1b[0m`);
+          return `${testName} (Failed in ${timeTaken} ms) - Error: ${error}`;
         }
       }
     }
-  );
+  };
 
-  const results = await Promise.all(testPromises);
+  const results: string[] = [];
+
+  for (let i = 0; i < testsToRun.length; i += CHUNK_SIZE) {
```

Comment:
Nice work ❇️ 

Still, it makes me sad we're re-implementing so much basic test framework functionality that'll get deleted when we convert these tests to run with Jest instead :(. Maybe we can get that work done during the next maintenance sprint that Chris was talking about today